### PR TITLE
Check if AP exists before adding it

### PR DESCRIPTION
**Pull request by dav1901**
Check if AP exists before adding it with WiFiMulti.addAP as suggested in #4253.

### CODE REVIEW

File: libraries/ESP8266WiFi/src/ESP8266WiFiMulti.cpp [hunk 1]
```diff
@@ -38,6 +38,10 @@ bool ESP8266WiFiMulti::addAP(const char* ssid, const char *passphrase) {
     return APlistAdd(ssid, passphrase);
 }
 
+bool ESP8266WiFiMulti::existsAP(const char* ssid, const char *passphrase) {
+    return APlistExists(ssid, passphrase);
+}
+
```
Comment (d-a-v) [inline libraries/ESP8266WiFi/src/ESP8266WiFiMulti.cpp]: What about only one `public: bool existsAP()` ?
(without the other private one)

Comment (dav1901) [reply libraries/ESP8266WiFi/src/ESP8266WiFiMulti.cpp]: I tried to make it similar to adding an AP, `public: bool addAP()` , `private: bool APlistAdd()`
Do you want me to change it or leave it as is?

Comment (d-a-v) [reply libraries/ESP8266WiFi/src/ESP8266WiFiMulti.cpp]: I think @devyte will enlighten us on that very subject

Comment (devyte) [reply libraries/ESP8266WiFi/src/ESP8266WiFiMulti.cpp]: @d-a-v is right: this is used to map different overloads to a single implementation in a private method, but in this case there is only one public method without overloads...
Sigh, IMHO leave it as-is, other public methods could be implemented with e.g.: String args.

**Pull request merged**